File: C++/PTwPZ KWIX.cpp
#include <stack>
#include <regex>
#include <string>
#include <iostream>

void printIndent(const int& depth) 
{
    for (int i = 0; i < depth * 4; i++) std::cout << ' ';
}

int main() 
{
    std::string line;
    std::stack<std::string> elements;
    std::regex tagRegex("<(\\w+)([^>]*)>");
    std::regex attributeRegex("(\\w+)=\"([^\"]*)\"");
    std::smatch match;
    int depth = 0;
    while (std::getline(std::cin, line)) 
    {
        auto it = line.cbegin();
        while (std::regex_search(it, line.cend(), match, tagRegex)) 
        {
            std::string tagName = match[1];
            std::string attributes = match[2];
            printIndent(depth);
            std::cout << tagName << ":\n";
            depth++;
            std::smatch attrMatch;
            auto attrIt = attributes.cbegin();
            while (std::regex_search(attrIt, attributes.cend(), attrMatch, attributeRegex)) 
            {
                printIndent(depth);
                std::cout << attrMatch[1] << " = " << attrMatch[2] << '\n';
                attrIt = attrMatch.suffix().first;
            }
            it = match.suffix().first;
            if (line.find("</" + tagName + ">") != std::string::npos) depth--;
        }
    }
    return 0;
}

File: Python/PTwPZ KWIX.py
import re
import sys

def print_indent(depth):
    print(" " * (depth * 4), end="")

def xml_to_kwix(xml_lines):
    tag_regex = re.compile(r"<(\w+)([^>]*)>")
    attr_regex = re.compile(r'(\w+)="([^"]*)"')
    depth = 0
    for line in xml_lines:
        matches = tag_regex.finditer(line)
        for match in matches:
            tag_name = match.group(1)
            attributes = match.group(2)
            print_indent(depth)
            print(f"{tag_name}:")
            depth += 1
            attr_matches = attr_regex.finditer(attributes)
            for attr_match in attr_matches:
                print_indent(depth)
                print(f"{attr_match.group(1)} = {attr_match.group(2)}")
            if f"</{tag_name}>" in line:
                depth -= 1

xml_to_kwix(sys.stdin.read().splitlines())